test/cases/coerced_tests.rb

```diff
@@ -51,6 +51,56 @@ def test_value_limit_violations_are_translated_to_specific_exception_coerced
   end
 end
 
+module ActiveRecord
+  class AdapterTestWithoutTransaction < ActiveRecord::TestCase
+    # SQL Server does not allow truncation of tables that are referenced by foreign key
+    # constraints. So manually remove/add foreign keys in test.
+    coerce_tests! :test_truncate_tables, :test_truncate_tables_with_query_cache
+    def test_truncate_tables_coerced
+      # Remove foreign key constraint to allow truncation.
+      @connection.remove_foreign_key :authors, :author_addresses
```

Comment:
Can't you use connection#disable_referential_integrity instead?

Reply:
No using `connection#disable_referential_integrity` does not work. The `disable_referential_integrity` method calls `ALTER TABLE [table] NOCHECK CONSTRAINT ALL` on the database but as you can see in the https://stackoverflow.com/a/3843826 and https://dba.stackexchange.com/a/190074 disabling constraints still does not allow tables to be truncated that have foreign key constraints.